modules/node_modules/@ncigdc/components/Oncogrid/dataMapping.js

```diff
@@ -115,50 +115,52 @@ export type TOccurence = {
   functionalImpact: string,
 };
 
-function expandObs(geneIdToSymbol) {
-  return ({ ssm, case: casObj = {} }): Array<TOccurence> => ssm.consequence
-    .map(c => c.transcript)
-    .map((consequence): TOccurence => {
-      const { annotation = {}, gene = {}, consequence_type } = consequence;
-      const geneId = gene.gene_id;
-
-      return {
-        // required
-        id: ssm.ssm_id,
-        donorId: casObj.case_id,
-        geneId,
-        consequence: consequence_type,
-
-        // optional
-        geneSymbol: geneIdToSymbol[geneId],
-        functionalImpact: annotation.impact,
-      };
-    });
-}
-
-export const mapOccurences = (
+type TMapOccurences = (
   occurrences: Array<TOccurenceInput>,
   donors: Array<TDonor>,
   genes: Array<TGene>,
   consequenceTypes: Array<string>,
   impacts: Array<string>
-): Array<TOccurence> => {
-  const donorIds: Array<string> = donors.map((d: TDonor): string => d.id);
-  const geneIds: Array<string> = genes.map((g: TGene): string => g.id);
+) => Array<TOccurence>;
+
+export const mapOccurences: TMapOccurences = (occurrences, donors, genes, consequenceTypes = [], impacts) => {
+  /* eslint-disable fp/no-loops, fp/no-let, fp/no-mutation, fp/no-mutating-methods */
+  const caseIds: Array<string> = donors.map((d: TDonor): string => d.id);
   const geneIdToSymbol = genes.reduce((acc, g: TGene) => ({ ...acc, [g.id]: g.symbol }), {});
 
-  function validOnco(occurrence: TOccurence): boolean {
-    return (
-      ((consequenceTypes || []).includes(occurrence.consequence)) &&
-      geneIds.indexOf(occurrence.geneId) >= 0 &&
-      donorIds.indexOf(occurrence.donorId) >= 0 &&
-      (!impacts.length || impacts.includes(occurrence.functionalImpact))
-    );
+  const arr = [];
+  for (let i = 0; i < occurrences.length; i += 1) {
+    const { ssm: { consequence, ssm_id }, case: { case_id } = {} } = occurrences[i];
+
+    if (caseIds.includes(case_id)) {
```

Comment:
Since you're already iterating over donors before this line, you can create a set or map instead, so that this can be a hash lookup instead of an includes iteration. I was trying to find some info about lookup time differences, but I think Set makes semantically the most sense and appears to be fast by some measure.

http://stackoverflow.com/questions/37464704/set-lookup-time-slower-than-object